azkaban-exec-server/src/main/java/azkaban/execapp/FlowExecutionMonitorManager.java

```diff
@@ -0,0 +1,116 @@
+package azkaban.execapp;
+
+import azkaban.sla.SlaOption;
+import azkaban.trigger.Condition;
+import azkaban.trigger.ConditionChecker;
+import azkaban.trigger.TriggerAction;
+import azkaban.trigger.builtin.KillExecutionAction;
+import azkaban.trigger.builtin.SlaAlertAction;
+import azkaban.trigger.builtin.SlaChecker;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.PriorityBlockingQueue;
+import org.apache.log4j.Logger;
+
+
+public class FlowExecutionMonitorManager extends Thread {
+  private static Logger logger = Logger.getLogger(FlowExecutionMonitorManager.class);
+  private BlockingQueue<Trigger> triggers = null;
+  private static final Duration CHECK_INTERVAL = Duration.ofMillis(15 * 1000);
```

Comment:
sounds good. btw. you have all sorts of helper methods - ofDays, ofHours, ofMinutes, etc.